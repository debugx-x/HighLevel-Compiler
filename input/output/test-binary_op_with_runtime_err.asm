        Jump         $$main                    
        DLabel       $eat-location-zero        
        DataZ        8                         
        DLabel       $print-format-integer     
        DataC        37                        %% "%d"
        DataC        100                       
        DataC        0                         
        DLabel       $print-format-float       
        DataC        37                        %% "%f"
        DataC        102                       
        DataC        0                         
        DLabel       $print-format-boolean     
        DataC        37                        %% "%s"
        DataC        115                       
        DataC        0                         
        DLabel       $print-format-character   
        DataC        37                        %% "%c"
        DataC        99                        
        DataC        0                         
        DLabel       $print-format-string      
        DataC        37                        %% "%s"
        DataC        115                       
        DataC        0                         
        DLabel       $print-format-newline     
        DataC        10                        %% "\n"
        DataC        0                         
        DLabel       $print-format-tab         
        DataC        9                         %% "\t"
        DataC        0                         
        DLabel       $print-format-space       
        DataC        32                        %% " "
        DataC        0                         
        DLabel       $boolean-true-string      
        DataC        116                       %% "true"
        DataC        114                       
        DataC        117                       
        DataC        101                       
        DataC        0                         
        DLabel       $boolean-false-string     
        DataC        102                       %% "false"
        DataC        97                        
        DataC        108                       
        DataC        115                       
        DataC        101                       
        DataC        0                         
        DLabel       $errors-general-message   
        DataC        82                        %% "Runtime error: %s\n"
        DataC        117                       
        DataC        110                       
        DataC        116                       
        DataC        105                       
        DataC        109                       
        DataC        101                       
        DataC        32                        
        DataC        101                       
        DataC        114                       
        DataC        114                       
        DataC        111                       
        DataC        114                       
        DataC        58                        
        DataC        32                        
        DataC        37                        
        DataC        115                       
        DataC        10                        
        DataC        0                         
        Label        $$general-runtime-error   
        PushD        $errors-general-message   
        Printf                                 
        Halt                                   
        DLabel       $errors-int-divide-by-zero 
        DataC        105                       %% "integer divide by zero"
        DataC        110                       
        DataC        116                       
        DataC        101                       
        DataC        103                       
        DataC        101                       
        DataC        114                       
        DataC        32                        
        DataC        100                       
        DataC        105                       
        DataC        118                       
        DataC        105                       
        DataC        100                       
        DataC        101                       
        DataC        32                        
        DataC        98                        
        DataC        121                       
        DataC        32                        
        DataC        122                       
        DataC        101                       
        DataC        114                       
        DataC        111                       
        DataC        0                         
        Label        $$i-divide-by-zero        
        PushD        $errors-int-divide-by-zero 
        Jump         $$general-runtime-error   
        DLabel       $errors-float-divide-by-zero 
        DataC        102                       %% "float divide by zero"
        DataC        108                       
        DataC        111                       
        DataC        97                        
        DataC        116                       
        DataC        32                        
        DataC        100                       
        DataC        105                       
        DataC        118                       
        DataC        105                       
        DataC        100                       
        DataC        101                       
        DataC        32                        
        DataC        98                        
        DataC        121                       
        DataC        32                        
        DataC        122                       
        DataC        101                       
        DataC        114                       
        DataC        111                       
        DataC        0                         
        Label        $$f-divide-by-zero        
        PushD        $errors-float-divide-by-zero 
        Jump         $$general-runtime-error   
        DLabel       $usable-memory-start      
        DLabel       $global-memory-block      
        DataZ        0                         
        Label        $$main                    
        PushI        2                         
        PushI        2                         
        Add                                    
        PushD        $print-format-integer     
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushF        2.000000                  
        PushF        2.000000                  
        FAdd                                   
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushF        100.000000                
        PushF        10.000000                 
        FAdd                                   
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushI        2                         
        PushI        2                         
        Subtract                               
        PushD        $print-format-integer     
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushF        2.000000                  
        PushF        2.000000                  
        FSubtract                              
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushF        100.000000                
        PushF        10.000000                 
        FSubtract                              
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushI        2                         
        PushI        2                         
        Multiply                               
        PushD        $print-format-integer     
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushF        2.000000                  
        PushF        2.000000                  
        FMultiply                              
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushF        100.000000                
        PushF        10.000000                 
        FMultiply                              
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushI        2                         
        PushI        2                         
        Label        -divide-1-2-stack         
        Duplicate                              
        JumpFalse    -divide-1-zero            
        Jump         -divide-1-join            
        Label        -divide-1-zero            
        Jump         $$i-divide-by-zero        
        Label        -divide-1-join            
        Divide                                 
        PushD        $print-format-integer     
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushF        2.000000                  
        PushF        2.000000                  
        Label        -fdivide-2-2-stack        
        Duplicate                              
        JumpFZero    -fdivide-2-zero           
        Jump         -fdivide-2-join           
        Label        -fdivide-2-zero           
        Jump         $$f-divide-by-zero        
        Label        -fdivide-2-join           
        FDivide                                
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushF        100.000000                
        PushF        10.000000                 
        Label        -fdivide-3-2-stack        
        Duplicate                              
        JumpFZero    -fdivide-3-zero           
        Jump         -fdivide-3-join           
        Label        -fdivide-3-zero           
        Jump         $$f-divide-by-zero        
        Label        -fdivide-3-join           
        FDivide                                
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushF        2.000000                  
        PushF        0.000000                  
        Label        -fdivide-4-2-stack        
        Duplicate                              
        JumpFZero    -fdivide-4-zero           
        Jump         -fdivide-4-join           
        Label        -fdivide-4-zero           
        Jump         $$f-divide-by-zero        
        Label        -fdivide-4-join           
        FDivide                                
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        Halt                                   
