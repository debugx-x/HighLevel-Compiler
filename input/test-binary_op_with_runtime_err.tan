main {
	print 2 + 2 \n;
	print 2.0 + 2.0 \n;
	print 1.0E+2 + 1.0E+1 \n;
	print 2 - 2 \n;
	print 2.0 - 2.0 \n;
	print 1.0E+2 - 1.0E+1 \n;
	print 2 * 2 \n;
	print 2.0 * 2.0 \n;
	print 1.0E+2 * 1.0E+1 \n;
	print 2 / 2 \n;
	print 2.0 / 2.0 \n;
	print 1.0E+2 / 1.0E+1 \n;
	print 2.0 / 0.0 \n;
}