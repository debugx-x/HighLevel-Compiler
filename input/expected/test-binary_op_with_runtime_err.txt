4
4.000000
110.000000
0
0.000000
90.000000
4
4.000000
1000.000000
1
1.000000
10.000000
Runtime error: float divide by zero
